analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyNullableValueAccessBase.cs

```diff
@@ -34,19 +34,24 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
             && reference.Property.Name == nameof(Nullable<int>.Value)
             && reference.Instance is { } instance
             && instance.Type.IsNullableValueType()
-            && context.HasConstraint(instance, ObjectConstraint.Null))
+            && context.HasConstraint(instance, ObjectConstraint.Null)
+            && NotNullFlowState(reference.Instance))
         {
             ReportIssue(instance, instance.Syntax.ToString());
         }
         else if (operationInstance.Kind == OperationKindEx.Conversion
             && operationInstance.ToConversion() is var conversion
             && conversion.Operand.Type.IsNullableValueType()
             && conversion.Type.IsNonNullableValueType()
-            && context.HasConstraint(conversion.Operand, ObjectConstraint.Null))
+            && context.HasConstraint(conversion.Operand, ObjectConstraint.Null)
+            && NotNullFlowState(conversion.Operand))
         {
             ReportIssue(conversion.Operand, conversion.Operand.Syntax.ToString());
         }
 
         return context.State;
+
+        bool NotNullFlowState(IOperation reference) =>
```

Comment:
@zsolt-kolbay-sonarsource 
As discussed offline, the local function name has been changed to `NotNullFlowState` from `NoNullForgiving`, since Roslyn flow analysis can deduce a non-null nullability state even in absence of the null forgiving operator.
On the other hand, when we reach this condition of the if statement, we are in a very narrow scenario, involving a nullable value type, and it's not clear to me how that would happen.